tests/test_util.rs

```diff
@@ -27,8 +27,18 @@ macro_rules! lowering_error {
             chalk_solve::SolverChoice::default(),
         )
         .checked_program()
-        .unwrap_err();
-        let expected = $expected;
-        assert_eq!(error.to_string(), expected.to_string());
+        .unwrap_err()
+        .to_string();
+        let expected = $expected.to_string();
+        crate::test_util::assert_same(&error, &expected);
```

Comment:
Shouldn't need the full path here, right?

Reply:
Due to macro hygiene the expanded code only has access to the scope of the caller, so we can either use a full path or bring `assert_same` into scope in the calling code. I tend towards the former for portability, but I can change it if that's preferable.

Reply:
Ah yeah, macros. This is better. Thanks :)